modules/jolt_physics/SCsub

```diff
@@ -0,0 +1,187 @@
+    "Jolt/Physics/Constraints/PathConstraintPath.cpp",
+    "Jolt/Physics/Constraints/PathConstraintPathHermite.cpp",
+    "Jolt/Physics/Constraints/PointConstraint.cpp",
+    "Jolt/Physics/Constraints/PulleyConstraint.cpp",
+    "Jolt/Physics/Constraints/RackAndPinionConstraint.cpp",
+    "Jolt/Physics/Constraints/SixDOFConstraint.cpp",
+    "Jolt/Physics/Constraints/SliderConstraint.cpp",
+    "Jolt/Physics/Constraints/SpringSettings.cpp",
+    "Jolt/Physics/Constraints/SwingTwistConstraint.cpp",
+    "Jolt/Physics/Constraints/TwoBodyConstraint.cpp",
+    "Jolt/Physics/Ragdoll/Ragdoll.cpp",
+    "Jolt/Physics/SoftBody/SoftBodyCreationSettings.cpp",
+    "Jolt/Physics/SoftBody/SoftBodyMotionProperties.cpp",
+    "Jolt/Physics/SoftBody/SoftBodyShape.cpp",
+    "Jolt/Physics/SoftBody/SoftBodySharedSettings.cpp",
+    "Jolt/Physics/Vehicle/MotorcycleController.cpp",
+    "Jolt/Physics/Vehicle/TrackedVehicleController.cpp",
+    "Jolt/Physics/Vehicle/VehicleAntiRollBar.cpp",
+    "Jolt/Physics/Vehicle/VehicleCollisionTester.cpp",
+    "Jolt/Physics/Vehicle/VehicleConstraint.cpp",
+    "Jolt/Physics/Vehicle/VehicleController.cpp",
+    "Jolt/Physics/Vehicle/VehicleDifferential.cpp",
+    "Jolt/Physics/Vehicle/VehicleEngine.cpp",
+    "Jolt/Physics/Vehicle/VehicleTrack.cpp",
+    "Jolt/Physics/Vehicle/VehicleTransmission.cpp",
+    "Jolt/Physics/Vehicle/Wheel.cpp",
+    "Jolt/Physics/Vehicle/WheeledVehicleController.cpp",
+    "Jolt/Renderer/DebugRenderer.cpp",
+    "Jolt/Renderer/DebugRendererPlayback.cpp",
+    "Jolt/Renderer/DebugRendererRecorder.cpp",
+    "Jolt/Renderer/DebugRendererSimple.cpp",
+    "Jolt/Skeleton/SkeletalAnimation.cpp",
+    "Jolt/Skeleton/Skeleton.cpp",
+    "Jolt/Skeleton/SkeletonMapper.cpp",
+    "Jolt/Skeleton/SkeletonPose.cpp",
+    "Jolt/TriangleGrouper/TriangleGrouperClosestCentroid.cpp",
+    "Jolt/TriangleGrouper/TriangleGrouperMorton.cpp",
+    "Jolt/TriangleSplitter/TriangleSplitter.cpp",
+    "Jolt/TriangleSplitter/TriangleSplitterBinning.cpp",
+    "Jolt/TriangleSplitter/TriangleSplitterFixedLeafSize.cpp",
+    "Jolt/TriangleSplitter/TriangleSplitterLongestAxis.cpp",
+    "Jolt/TriangleSplitter/TriangleSplitterMean.cpp",
+    "Jolt/TriangleSplitter/TriangleSplitterMorton.cpp",
+    # We don't need ObjectStream/RTTI, so we omit those files. See `ENABLE_OBJECT_STREAM` in `Jolt/Jolt.cmake` upstream.
+    # "Jolt/ObjectStream/ObjectStream.cpp",
+    # "Jolt/ObjectStream/ObjectStreamBinaryIn.cpp",
+    # "Jolt/ObjectStream/ObjectStreamBinaryOut.cpp",
+    # "Jolt/ObjectStream/ObjectStreamIn.cpp",
+    # "Jolt/ObjectStream/ObjectStreamOut.cpp",
+    # "Jolt/ObjectStream/ObjectStreamTextIn.cpp",
+    # "Jolt/ObjectStream/ObjectStreamTextOut.cpp",
+    # "Jolt/ObjectStream/TypeDeclarations.cpp",
+]
+
+thirdparty_sources = [thirdparty_dir + file for file in thirdparty_sources]
+
+env_jolt.Prepend(CPPPATH=[thirdparty_dir])
+
+if env.dev_build:
+    env_jolt.Append(CPPDEFINES=["JPH_ENABLE_ASSERTS"])
```

Comment:
Should we really enable Jolt asserts on dev builds? I understand the rationale that in a dev build we might want all debug features enabled, but in practice the vast majority of users of dev builds won't be Jolt experts and won't have much use for it crashing on them.

For this kind of stuff, in my experience it's used occasionally on demand by the feature's maintainer to debug specific issues, in which case it could be made a module-specific SCons argument (see `modules/minimp3/config.py` for an example of that).

I'm not too strongly opinionated about it though, and we can start with it enabled in all dev builds and see if it turns out to be a bother for engine contributors.

Reply:
The assert is implemented in `jolt_globals.cpp` as an `ERR_PRINT`, as seen [here](https://github.com/mihe/godot/blob/828ab604c101616f1a2f27f85a1af39f0021cf8b/modules/jolt_physics/jolt_globals.cpp#L78-L81), so it doesn't crash anything per se.

I guess the point still stands that most people won't know what to do about the error once they do see it printed, but I've found that it's easy to think that everything is fine only to fire up a dev build and see Jolt trigger a bunch of asserts because something somewhere is ill-formed, so my vote would probably be to leave this enabled until it's found to be too annoying.

Reply:
Sounds good to me.